http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -119,12 +124,35 @@ private EndpointResponse handleTimeout(Throwable throwable) {
             endpointRequest.getPath())
         .timeout(httpActionOptions.getRequestTimeoutMs());
 
+    ResponsePredicate noApplicationJson = ResponsePredicate
+        .create(ResponsePredicate.JSON, result -> {
+          throw new ReplyException(ReplyFailure.RECIPIENT_FAILURE, result.message());
+        });
+    if (httpActionOptions.getResponseOptions().getPredicates().contains(JSON)) {
+      request.expect(io.vertx.reactivex.ext.web.client.predicate.ResponsePredicate
+          .newInstance(noApplicationJson));
+    }
+    attachResponsePredicatesToRequest(request,
+        httpActionOptions.getResponseOptions().getPredicates());
     endpointRequest.getHeaders().entries()
         .forEach(entry -> request.putHeader(entry.getKey(), entry.getValue()));
 
     return request.rxSend();
   }
 
+  private void attachResponsePredicatesToRequest(HttpRequest<Buffer> request,
+      Set<String> predicates) {
+    predicates.forEach(predicate -> {
+      if (!JSON.equals(predicate)) {
```

Comment:
I don't understand the logic behind this function.
Why do you treat `JSON` predicate separately?

Reply:
I'm treating it separately because other predicates we can simply attach to request and if something will happen with them, the default behavior will be fired. In the case of JSON predicate, we want to add some custom behavior which is throwing ReplyException. Predicates are attached to the request basing on the configuration - if JSON appears there we have to treat them separately to add this behavior because if we won't do it, it will be attached to the request in the same way as other predicates and will behave in the same way as other predicates - it won't throw ReplyException